akka-stream-typed/src/test/java/docs/akka/stream/typed/ActorSourceWithBackpressureExample.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2018-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package docs.akka.stream.typed;
+
+// #sample
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+import akka.stream.CompletionStrategy;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.stream.typed.javadsl.ActorSource;
+
+import java.util.Optional;
+
+class StreamFeeder extends AbstractBehavior<StreamFeeder.Emitted> {
+  /** Signals that the latest element is emitted into the stream */
+  static final class Emitted {}
+
+  public interface Event {}
+
+  final class Element implements Event {
+    private final String content;
+
+    public Element(String content) {
+      this.content = content;
+    }
+
+    @Override
+    public String toString() {
+      return "Element(" + content + ")";
+    }
+  }
+
+  final class ReachedEnd implements Event {}
+
+  final class FailureOccured implements Event {
+    private final Exception ex;
+
+    public FailureOccured(Exception ex) {
+      this.ex = ex;
+    }
+  }
+
+  public static Behavior<Emitted> create() {
+    return Behaviors.setup(StreamFeeder::new);
+  }
+
+  private int counter = 0;
+  private final ActorRef<Event> streamSource;
+
+  private StreamFeeder(ActorContext<Emitted> context) {
+    super(context);
+    streamSource = runStream(context.getSelf(), context.getSystem());
+    streamSource.tell(new Element("first"));
+  }
+
+  @Override
+  public Receive<Emitted> createReceive() {
+    return newReceiveBuilder().onMessage(Emitted.class, this::onEmitted).build();
+  }
+
+  private ActorRef<Event> runStream(ActorRef<Emitted> ackReceiver, ActorSystem<?> system) {
```

Comment:
good practice is to define Future and Stream stuff outside of the actor Behavior to avoid accidental access to actor state. `static` would do. Pass in what is needed as parameter, such as the self ActorRef